pkg/sidecar-controller/snapshot_controller.go

```diff
@@ -317,34 +307,66 @@ func (ctrl *csiSnapshotSideCarController) createSnapshotOperation(content *crdv1
 		return content, nil
 	}
 
+	return ctrl.createSnapshotWrapper(content)
+}
+
+// This is a wrapper function for the snapshot creation process.
+func (ctrl *csiSnapshotSideCarController) createSnapshotWrapper(content *crdv1.VolumeSnapshotContent) (*crdv1.VolumeSnapshotContent, error) {
+	klog.Infof("createSnapshotWrapper: Creating snapshot for content %s through the plugin ...", content.Name)
+
 	class, snapshotterCredentials, err := ctrl.getCSISnapshotInput(content)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get input parameters to create snapshot for content %s: %q", content.Name, err)
 	}
 
+	// NOTE(xyang): handle create timeout
+	// Add an annotation to indicate the snapshot creation request has been
+	// sent to the storage system and the controller is waiting for a response.
+	// The annotation will be removed after the storage system has responded with
+	// success or permanent failure. If the request times out, annotation will
+	// remain on the content to avoid potential leaking of a snapshot resource on
+	// the storage system.
+	err = ctrl.setAnnVolumeSnapshotBeingCreated(content)
```

Comment:
I will share one of my concerns here w.r.t the annotation's life cycle but I do not really have a good solution.
The annotation will be added whenever reached here, however only will it be removed when CSI returns a permanent failure or success in the end of this function.
I'd suggest adding a check at this [line](https://github.com/kubernetes-csi/external-snapshotter/blob/8b8a4d81aeddc9b2e0b229e7ac516030f7f697ff/pkg/sidecar-controller/snapshot_controller.go#L85) before return, if such an annotation exists, just remove it when status.ReadyToUse == true